### PR TITLE
Update specs and export

### PR DESCRIPTION
**Pull request by martinsumner**
Add exported specs for all functions required as inputs

### CODE REVIEW

File: src/aae_exchange.erl [hunk 1]
```diff
@@ -237,8 +245,8 @@ start(BlueList, PinkList, RepairFun, ReplyFun) ->
 
 -spec start(exchange_type(),
             input_list(), input_list(),
-            fun((list({any(), any(), any()})) -> ok),
-            fun(({closing_state(), non_neg_integer()}) -> ok),
+            repair_fun(),
```
Comment (ThomasArts) [inline src/aae_exchange.erl]: This type is different ?? from a single 3-tuple to a 2-tuple of 2-tuples. Is that on purpose?

File: src/aae_exchange.erl [hunk 2]
```diff
@@ -801,7 +809,7 @@ compare_branches(BlueBranches, PinkBranches) ->
         end,
     lists:foldl(FoldFun, [], lists:seq(1, length(BlueBranches))).
 
--spec compare_clocks(list(tuple()), list(tuple())) -> list(tuple()).
+-spec compare_clocks(list(tuple()), list(tuple())) -> list(repair_input()).
 %% @doc
 %% Find the differences between the lists - and return a list of
 %% {B, K, blue-side VC, pink-side VC}
```
Comment (ThomasArts) [inline src/aae_exchange.erl]: Here in the comment we return a 4-tuple, but repair-input is a tuple of 2-tuples.

File: src/aae_exchange.erl (hunk 1 above)
Comment (martinsumner) [reply src/aae_exchange.erl]: sorry, I got the original type wrong (although dialyzer didn't seem to notice).  The new one is correct - it is a 2-tuple of 2-tuples

File: src/aae_exchange.erl (hunk 2 above)
Comment (martinsumner) [reply src/aae_exchange.erl]: the comment is wrong, I will correct

**Pull request merged**